--- a/net/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityMinecartIdentifiersFix.java
@@ -45,7 +45,7 @@
    
                   return Pair.of(
                      s,
-                     typed.write()
+                     ((com.mojang.serialization.DataResult<Dynamic<Pair<String, ?>>>)typed.write())
                         .map(p_145294_ -> taggedchoicetype1.types().get(s).read(p_145294_))
                         .result()
                         .orElseThrow(() -> new IllegalStateException("Could not read the new minecart."))
