--- a/net/minecraft/data/loot/BlockLootSubProvider.java
+++ b/net/minecraft/data/loot/BlockLootSubProvider.java
@@ -132,7 +132,7 @@
    }
 
    protected LootTable.Builder m_245514_(Block p_249305_, ItemLike p_251905_) {
-      return m_247502_(p_249305_, this.m_247733_(p_249305_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(p_251905_)));
+      return m_247502_(p_249305_, this.m_247733_(p_249305_, LootItem.m_79579_(p_251905_)));
    }
 
    protected LootTable.Builder m_245765_(ItemLike p_251584_, NumberProvider p_249865_) {
@@ -345,7 +345,7 @@
          p_252139_,
          this.m_246108_(
             p_252139_,
-            ((LootPoolSingletonContainer.Builder)LootItem.m_79579_(Items.f_42404_).m_79080_(LootItemRandomChanceCondition.m_81927_(0.125F)))
+            LootItem.m_79579_(Items.f_42404_).m_79080_(LootItemRandomChanceCondition.m_81927_(0.125F))
                .m_79078_(ApplyBonusCount.m_79921_(Enchantments.f_44987_, 2))
          )
       );
@@ -397,7 +397,7 @@
                .m_79076_(
                   this.m_246108_(
                      p_249088_,
-                     ((LootPoolSingletonContainer.Builder)((LootPoolSingletonContainer.Builder)LootItem.m_79579_(p_249088_).m_79080_(p_251535_))
+                     LootItem.m_79579_(p_249088_).m_79080_(p_251535_)
                            .m_230987_(
                               Direction.values(),
                               p_251536_ -> SetItemCountFunction.m_165414_(ConstantValue.m_165692_(1.0F), true)
@@ -405,7 +405,7 @@
                                        LootItemBlockStatePropertyCondition.m_81769_(p_249088_)
                                           .m_81784_(StatePropertiesPredicate.Builder.m_67693_().m_67703_(MultifaceBlock.m_153933_(p_251536_), true))
                                     )
-                           ))
+                           )
                         .m_79078_(SetItemCountFunction.m_165414_(ConstantValue.m_165692_(-1.0F), true))
                   )
                )
@@ -415,7 +415,7 @@
    protected LootTable.Builder m_246047_(Block p_250088_, Block p_250731_, float... p_248949_) {
       return m_246160_(
             p_250088_,
-            this.m_247733_(p_250088_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(p_250731_))
+            this.m_247733_(p_250088_, LootItem.m_79579_(p_250731_))
                .m_79080_(BonusLevelTableCondition.m_81517_(Enchantments.f_44987_, p_248949_))
          )
          .m_79161_(
@@ -436,7 +436,7 @@
                .m_165133_(ConstantValue.m_165692_(1.0F))
                .m_79080_(f_244248_)
                .m_79076_(
-                  this.m_247733_(p_249535_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(Items.f_42410_))
+                  this.m_247733_(p_249535_, LootItem.m_79579_(Items.f_42410_))
                      .m_79080_(BonusLevelTableCondition.m_81517_(Enchantments.f_44987_, 0.005F, 0.0055555557F, 0.00625F, 0.008333334F, 0.025F))
                )
          );
@@ -480,7 +480,7 @@
             .m_79078_(SetItemCountFunction.m_165412_(ConstantValue.m_165692_(2.0F)))
             .m_79080_(f_243905_))
          .m_7170_(
-            this.m_247733_(p_248590_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(Items.f_42404_))
+            this.m_247733_(p_248590_, LootItem.m_79579_(Items.f_42404_))
                .m_79080_(LootItemRandomChanceCondition.m_81927_(0.125F))
          );
       return LootTable.m_79147_()
