--- a/net/minecraft/data/loot/packs/VanillaBlockLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaBlockLoot.java
@@ -842,7 +842,7 @@
          p_250064_ -> LootTable.m_79147_()
                .m_79161_(
                   LootPool.m_79043_()
-                     .m_79076_(this.m_246108_(p_250064_, (FunctionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(Items.f_42726_)))
+                     .m_79076_(this.m_246108_(p_250064_, LootItem.m_79579_(Items.f_42726_)))
                )
                .m_79161_(
                   LootPool.m_79043_()
@@ -1088,7 +1088,7 @@
       this.m_246481_(
          Blocks.f_50033_,
          p_250546_ -> m_246160_(
-               p_250546_, this.m_247733_(p_250546_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(Items.f_42401_))
+               p_250546_, this.m_247733_(p_250546_, LootItem.m_79579_(Items.f_42401_))
             )
       );
       this.m_246481_(
@@ -1119,7 +1119,7 @@
                   LootPool.m_79043_()
                      .m_165133_(ConstantValue.m_165692_(1.0F))
                      .m_79076_(
-                        this.m_247733_(p_249769_, (ConditionUserBuilder<LootPoolSingletonContainer.Builder<?>>)LootItem.m_79579_(p_249769_))
+                        this.m_247733_(p_249769_, LootItem.m_79579_(p_249769_))
                            .m_79080_(LootItemEntityPropertyCondition.m_81862_(LootContext.EntityTarget.THIS))
                      )
                )
