--- a/net/minecraft/server/level/ChunkTaskPriorityQueue.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueue.java
@@ -94,7 +94,7 @@
             this.f_140510_++;
          }
 
-         return list.stream().map(p_140529_ -> p_140529_.map(Either::left).orElseGet(() -> Either.right(this.m_140525_(j))));
+         return list.stream().map(p_140529_ -> p_140529_.<Either<T, Runnable>>map(Either::left).orElseGet(() -> Either.right(this.m_140525_(j))));
       }
    }
 
