--- a/net/minecraft/commands/arguments/blocks/BlockInput.java
+++ b/net/minecraft/commands/arguments/blocks/BlockInput.java
@@ -39,7 +39,7 @@
          return false;
       } else {
          for (Property<?> property : this.f_114663_) {
-            if (blockstate.m_61143_((Property<boolean>)property) != this.f_114662_.m_61143_((Property<boolean>)property)) {
+            if (blockstate.m_61143_(property) != this.f_114662_.m_61143_(property)) {
                return false;
             }
          }
