--- a/net/minecraft/world/item/HoneycombItem.java
+++ b/net/minecraft/world/item/HoneycombItem.java
@@ -88,7 +_,7 @@
                 level.playSound(player, pos, SoundEvents.HONEYCOMB_WAX_ON, SoundSource.BLOCKS, 1.0F, 1.0F);
             }
 
-            return InteractionResult.SUCCESS;
+            return (InteractionResult)InteractionResult.SUCCESS;
         }).orElse(InteractionResult.PASS);
     }
 
