--- a/net/minecraft/advancements/Criterion.java
+++ b/net/minecraft/advancements/Criterion.java
@@ -18,7 +18,7 @@
       if (criteriontrigger == null) {
          throw new JsonSyntaxException("Invalid criterion trigger: " + resourcelocation);
       } else {
-         return m_300333_(p_11418_, p_11419_, (CriterionTrigger<T>)criteriontrigger);
+         return m_300333_(p_11418_, p_11419_, criteriontrigger);
       }
    }
 
