--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -582,9 +582,9 @@
          if (p_269770_.isPresent()) {
             ChunkAccess chunkaccess = ChunkSerializer.m_188230_(this.f_140133_, this.f_140138_, p_140418_, p_269770_.get());
             this.m_140229_(p_140418_, chunkaccess.m_6415_().m_62494_());
-            return Either.left(chunkaccess);
+            return Either.<ChunkAccess, ChunkHolder.ChunkLoadingFailure>left(chunkaccess);
          } else {
-            return Either.left(this.m_214961_(p_140418_));
+            return Either.<ChunkAccess, ChunkHolder.ChunkLoadingFailure>left(this.m_214961_(p_140418_));
          }
       }, this.f_140135_).exceptionallyAsync(p_214888_ -> this.m_214901_(p_214888_, p_140418_), this.f_140135_);
    }
