--- a/net/minecraft/client/PeriodicNotificationManager.java
+++ b/net/minecraft/client/PeriodicNotificationManager.java
@@ -123,9 +123,9 @@
    }
 
    public static record Notification(long f_205328_, long f_205329_, String f_205330_, String f_205331_) {
-      public Notification(long f_205328_, long f_205330_, String f_205330_, String f_205331_) {
-         this.f_205328_ = f_205328_ != 0L ? f_205328_ : f_205330_;
-         this.f_205329_ = f_205330_;
+      public Notification(long f_205328_, long f_205329_, String f_205330_, String f_205331_) {
+         this.f_205328_ = f_205328_ != 0L ? f_205328_ : f_205329_;
+         this.f_205329_ = f_205329_;
          this.f_205330_ = f_205330_;
          this.f_205331_ = f_205331_;
       }
