--- a/net/minecraft/network/syncher/SynchedEntityData.java
+++ b/net/minecraft/network/syncher/SynchedEntityData.java
@@ -212,7 +212,7 @@
             )
          );
       } else {
-         p_135376_.m_135397_(p_254484_.f_252525_);
+         p_135376_.m_135397_((T)p_254484_.f_252525_);
       }
    }
 
@@ -284,7 +284,7 @@
          if (entitydataserializer == null) {
             throw new DecoderException("Unknown serializer type " + i);
          } else {
-            return m_252951_(p_254314_, p_254356_, (EntityDataSerializer<T>)entitydataserializer);
+            return m_252951_(p_254314_, p_254356_, entitydataserializer);
          }
       }
 
