--- a/net/minecraft/network/protocol/game/VecDelta.java
+++ b/net/minecraft/network/protocol/game/VecDelta.java
@@ -56,10 +_,8 @@
                     output.writeShort(ya);
                     output.writeShort(za);
                     break;
-                case VecDelta.Stepped(List var20):
-                    List var10 = var20;
-
-                    for (VecDelta.Stepped.DeltaStep step : var10) {
+                case VecDelta.Stepped(List<VecDelta.Stepped.DeltaStep> steps):
+                    for (VecDelta.Stepped.DeltaStep step : steps) {
                         output.writeVarInt(step.ticks);
                         output.writeShort(step.xa);
                         output.writeShort(step.ya);
