--- a/com/mojang/renderpearl/backend/vulkan/init/FeatureSet.java
+++ b/com/mojang/renderpearl/backend/vulkan/init/FeatureSet.java
@@ -15,6 +_,13 @@
 
 @OnlyIn(Dist.CLIENT)
 public record FeatureSet(String name, Set<String> extensions, Set<VulkanFeature> features, FeatureSet.Condition condition) {
+    public FeatureSet(String name, Set<String> extensions, Set<VulkanFeature> features, FeatureSet.Condition condition) {
+        this.name = name;
+        this.extensions = Set.copyOf(extensions);
+        this.features = Set.copyOf(features);
+        this.condition = condition;
+    }
+
     public FeatureSet(String name, Set<String> extensions, Set<VulkanFeature> features) {
         this(name, extensions, features, FeatureSet.Condition.IDENTITY_CONDITION);
     }
