--- a/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
+++ b/net/minecraft/world/level/levelgen/heightproviders/ConstantHeight.java
@@ -11,7 +11,7 @@
    public static final ConstantHeight f_161945_ = new ConstantHeight(VerticalAnchor.m_158922_(0));
    public static final Codec<ConstantHeight> f_161946_ = Codec.either(
          VerticalAnchor.f_158914_,
-         RecordCodecBuilder.create(
+         RecordCodecBuilder.<ConstantHeight>create(
             p_161955_ -> p_161955_.group(VerticalAnchor.f_158914_.fieldOf("value").forGetter(p_161967_ -> p_161967_.f_161947_))
                   .apply(p_161955_, ConstantHeight::new)
          )
