--- a/net/minecraft/core/RegistrySynchronization.java
+++ b/net/minecraft/core/RegistrySynchronization.java
@@ -67,7 +67,7 @@
    private static <K extends ResourceKey<? extends Registry<?>>, V extends Registry<?>> Codec<RegistryAccess> m_245940_(UnboundedMapCodec<K, V> p_249934_) {
       return p_249934_.xmap(
          RegistryAccess.ImmutableRegistryAccess::new,
-         p_251578_ -> m_247199_(p_251578_).collect(ImmutableMap.toImmutableMap(p_250395_ -> p_250395_.f_206233_(), p_248951_ -> p_248951_.f_206234_()))
+         p_251578_ -> m_247199_(p_251578_).collect(ImmutableMap.toImmutableMap(p_250395_ -> (K)p_250395_.f_206233_(), p_248951_ -> (V)p_248951_.f_206234_()))
       );
    }
 
