--- a/net/minecraft/commands/synchronization/ArgumentTypeInfos.java
+++ b/net/minecraft/commands/synchronization/ArgumentTypeInfos.java
@@ -142,7 +142,7 @@
       if (argumenttypeinfo == null) {
          throw new IllegalArgumentException(String.format(Locale.ROOT, "Unrecognized argument type %s (%s)", p_235383_, p_235383_.getClass()));
       } else {
-         return argumenttypeinfo;
+         return (ArgumentTypeInfo<A, ?>)argumenttypeinfo;
       }
    }
 
