--- a/net/minecraft/util/valueproviders/ConstantInt.java
+++ b/net/minecraft/util/valueproviders/ConstantInt.java
@@ -9,7 +9,7 @@
    public static final ConstantInt f_146476_ = new ConstantInt(0);
    public static final Codec<ConstantInt> f_146477_ = Codec.either(
          Codec.INT,
-         RecordCodecBuilder.create(
+         RecordCodecBuilder.<ConstantInt>create(
             p_146490_ -> p_146490_.group(Codec.INT.fieldOf("value").forGetter(p_146498_ -> p_146498_.f_146478_)).apply(p_146490_, ConstantInt::new)
          )
       )
